server/core/runtime/policy/conftest_executor.go

```diff
@@ -62,62 +55,61 @@ func NewConfTestExecutor(creator githubapp.ClientCreator, org string) *ConfTestE
 // filter that will filter out all policy failures based on the filter criteria.
 func (c *ConfTestExecutor) Run(_ context.Context, prjCtx command.ProjectContext, executablePath string, envs map[string]string, workdir string, extraArgs []string) (string, error) {
-	var policyArgs []Arg
 	var policyNames []string
 	var failedPolicies []valid.PolicySet
+	var totalCmdOutput string
+	var policyErr error
+
 	inputFile := filepath.Join(workdir, prjCtx.GetShowResultFileName())
 	scope := prjCtx.Scope.SubScope(conftestScope)
 
 	for _, policySet := range prjCtx.PolicySets.PolicySets {
-		path, err := c.SourceResolver.Resolve(policySet)
-		// Let's not fail the whole step because of a single failure. Log and fail silently
-		if err != nil {
-			prjCtx.Log.ErrorContext(prjCtx.RequestCtx, fmt.Sprintf("Error resolving policyset %s. err: %s", policySet.Name, err.Error()))
-			continue
+		var policyArgs []Arg
+		for _, path := range policySet.Paths {
+			policyArgs = append(policyArgs, NewPolicyArg(path))
 		}
-		policyArgs = append(policyArgs, NewPolicyArg(path))
 		policyNames = append(policyNames, policySet.Name)
-	}
-
-	args := ConftestTestCommandArgs{
-		PolicyArgs: policyArgs,
-		ExtraArgs:  extraArgs,
-		InputFile:  inputFile,
-		Command:    executablePath,
-	}
-	serializedArgs, err := args.build()
-	if err != nil {
-		prjCtx.Log.WarnContext(prjCtx.RequestCtx, "No policies have been configured")
-		scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
-		return "", errors.Wrap(err, "building args")
-	}
+		args := ConftestTestCommandArgs{
+			PolicyArgs: policyArgs,
+			ExtraArgs:  extraArgs,
+			InputFile:  inputFile,
+			Command:    executablePath,
+		}
+		serializedArgs, err := args.build()
+		if err != nil {
+			prjCtx.Log.WarnContext(prjCtx.RequestCtx, "No policies have been configured")
+			scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
+			return "", errors.Wrap(err, "building args")
+		}
 
-	// TODO: run each policy set separately and use each pass/failure decision to populate failedPolicies
-	cmdOutput, policyErr := c.Exec.CombinedOutput(serializedArgs, envs, workdir)
-	if policyErr != nil {
-		failedPolicies = prjCtx.PolicySets.PolicySets
+		cmdOutput, cmdErr := c.Exec.CombinedOutput(serializedArgs, envs, workdir)
+		// Continue running other policies if one fails since it might not be the only failing one
+		if cmdErr != nil {
+			policyErr = cmdErr
+			failedPolicies = append(failedPolicies, policySet)
+		}
+		totalCmdOutput = fmt.Sprintf("%s\n%s", totalCmdOutput, cmdOutput)
 	}
 
 	title := c.buildTitle(policyNames)
-	output := c.sanitizeOutput(inputFile, title+cmdOutput)
+	output := c.sanitizeOutput(inputFile, title+totalCmdOutput)
 	if prjCtx.InstallationToken == 0 {
 		prjCtx.Log.ErrorContext(prjCtx.RequestCtx, "missing installation token")
 		scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
 		return output, errors.New(internalError)
```

Comment:
nit: wrap the `internalError` with more context on the error, eg: missing installation token

Reply:
We also output the error line. I wanted to keep from exposing internal Atlantis errors to the user, hence logging the actual error and returning a generic internal error here.

Reply:
Ahh makes sense!